### PR TITLE
Simple text example

### PR DESCRIPTION
**Pull request by koober**
added basic text example
added link to example in docs/api/extras/core/Font.html
added link to example in docs/api/loaders/FontLoader.html

### CODE REVIEW

**Comment by koober** [comment]
Ref Issue #10725

File: examples/webgl_geometry_text_shapes.html [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+				renderer.setSize( window.innerWidth, window.innerHeight );
+				document.body.appendChild( renderer.domElement );
+
+				window.addEventListener( 'resize', onWindowResize, false );
+
+			} // end init
+
+			function xMove( shape, shapeMid ) {
+
+				return shape.applyMatrix( new THREE.Matrix4().makeTranslation( shapeMid, 0, 0 ) );
```
Comment (WestLangley) [inline examples/webgl_geometry_text_shapes.html]:
```javascript
return shape.translate( shapeMid, 0, 0 );
```
You might as well inline that method.

**Comment by WestLangley** [comment]
https://rawgit.com/koober/three.js/dev/examples/webgl_geometry_text_shapes.html

Nice 2D text example. Any idea why the example has 27 geometries and 26 draw calls?

File: examples/webgl_geometry_text_shapes.html (hunk 1 above)
Comment (koob3d) [inline examples/webgl_geometry_text_shapes.html]: Neat! Will do... thanks.

**Comment by koob3d** [comment]
Confession. I only use  console.log(renderer.info.render.calls), which says 27.
Suggestions for a better stats tool welcome.

**Comment by WestLangley** [comment]
This issue is not with the stats tool, unless it has a bug.

Why so many draw calls, why so many geometries, and why fewer draw calls than geometries?

File: examples/webgl_geometry_text_shapes.html [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+						var lineMesh = new THREE.Line( lineGeometry, matDark );
+						lineText.add( lineMesh );
+
+						if ( shape.holes && shape.holes.length > 0 ) {
+
+							for ( var j = 0; j < shape.holes.length; j ++ ) {
+
+								var hole = shape.holes[ j ];
+								shapes.push( hole );
```
Comment (WestLangley) [inline examples/webgl_geometry_text_shapes.html]: You are modifying the length of shapes as you iterate over it.

File: examples/webgl_geometry_text_shapes.html [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+					} );
+
+					var matLite = new THREE.MeshBasicMaterial( {
+						color: color,
+						transparent: true,
+						opacity: 0.4,
+						side: THREE.DoubleSide
+					} );
+
+					var xMove = function( shape, shapeMid ) {
```
Comment (WestLangley) [inline examples/webgl_geometry_text_shapes.html]: Like I said, you don't use the return value and you don't need a separate function for this; just inline the code.

File: examples/webgl_geometry_text_shapes.html [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+					for ( var i = 0; i < shapes.length; i ++ ) {
+
+						var shape = shapes[ i ];
+
+						var lineGeometry = shape.createPointsGeometry();
+
+						xMove( lineGeometry, xMid );
+
+						var lineMesh = new THREE.Line( lineGeometry, matDark );
```
Comment (WestLangley) [inline examples/webgl_geometry_text_shapes.html]: `Line` uses `LineBasicMaterial`, not `MeshBasicMaterial`.

**Comment by koober** [comment]
Yes, I see what he meant by hacky now.
Sorry, I didn't wish to imply fault in any stats tool. 
For a single drawcall on the line geometries, am I right to think they need to be converted to LineSegments and be merged? I can play with that for a while but not too sure of a result. 

Or I can remove the line text? Then rather have a full example it should just go in the docs page for Font, which doesn't have an example attributed to it currently

**Comment by WestLangley** [comment]
@koober I will leave it up to @mrdoob. Options are to (1) leave it as it is, (2) remove the line text, (3) do (2) and remove the example altogether, adding it to the docs page for Font, or (4) merge the line geometry. `BufferGeometry.merge()` works, but you have to pre-allocate the buffers.

`Geometry` merge is really inefficient for a large number of geometries. We could use this as an opportunity to show how to merge buffer geometries.

**Comment by mrdoob** [comment]
Maybe we could even turn this into `THREE.OutlineShapeGeometry`...?

**Comment by WestLangley** [comment]
>Maybe we could even turn this into THREE.OutlineShapeGeometry...?

Maybe. FWIW, the technique used in this PR is also used in http://threejs.org/examples/webgl_geometry_shapes.html.

**Comment by koober** [comment]
I looked at webgl_geometry_shapes before, but couldn't fit .generateShapes() in that format.
Anyway, I've updated that example to include holes, but also removed a couple of redundant lines.

**Comment by WestLangley** [comment]
NIce.

https://rawgit.com/koober/three.js/dev/examples/webgl_geometry_shapes.html (proposed)

https://rawgit.com/mrdoob/three.js/dev/examples/webgl_geometry_shapes.html (original)

Maybe `webgl_geometry_shapes.html` should be in a separate PR, though.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**